### PR TITLE
Fix in situ insertion and refactor database package

### PR DESCRIPTION
**Pull request by amehta-scottlogic**
## Changes

- fixes broken in_situ upsert due to wrong keys being used when upserting
- adds more fields to upsert for forecast data since last change

### CODE REVIEW

File: air-quality-backend/src/database/mongo_db_operations.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+def upsert_data(collection_name: str, keys: list[str], data):
+    if len(data) == 0:
+        return
+    collection = get_collection(collection_name)
+    now = datetime.utcnow()
+
+    update_operations = [
+        UpdateOne(
+            {key: doc[key] for key in keys},
```
Comment (amehta-scottlogic) [inline air-quality-backend/src/database/mongo_db_operations.py:22]: These are the keys we will upsert on if there is a document already in the collection

File: air-quality-backend/src/database/mongo_db_operations.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            [
+                {
+                    "$set": {
+                        "last_modified_time": now,
```
Comment (amehta-scottlogic) [inline air-quality-backend/src/database/mongo_db_operations.py:26]: Always add a last_modified_time property to be now

File: air-quality-backend/src/database/mongo_db_operations.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                        "created_time": {
+                            "$cond": [
+                                {"$not": ["$created_time"]},
+                                now,
+                                "$created_time",
+                            ]
+                        },
```
Comment (amehta-scottlogic) [inline air-quality-backend/src/database/mongo_db_operations.py:33]: Add created_time if not already there

File: air-quality-backend/src/database/mongo_db_operations.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                        **doc,
```
Comment (amehta-scottlogic) [inline air-quality-backend/src/database/mongo_db_operations.py:34]: Add rest of fields in document

**Review by mwalker-scottlogic — APPROVED** [review]
lgtm

**Pull request merged**